WIP: Manual Fixes. JuliaCon 2016 Edition

omus opened the pull request:
I've been re-reading the manual to discover any new features I may have missed over the last year. During my reading so far I've found some issues:
1. "Elementary Functions" section seemed very out of place. Mainly the problem is that it breaks up two sections that both talked about rounding
2.  The "Byte Array Literals" section contained a sentence talking about non-standard string literals conventions. I've moved it to a more appropriate area
3. Found a trivial typo in the "Version Number Literals" section

omus committed:
Move non-standard string literal conventions

omus committed:
Fix invalid version literal

omus committed:
Move "Elementary Function" section to improve flow

The original location of the "Elementary Function" section broke the
flow between "Numerical Conversions" about round and the next section:
"Rounding Functions".

Additionally moved the "Elementary Function" section to be a subsection
of "Numeric Comparisons".

omus committed:
Consistently refer to Julia over julia

The manual would sometimes refer to the language as either "Julia" or
"julia". This commit changes references to the Julia language as "Julia"
and any reference to the command line program as `julia`.

omus commented:
I noticed inconsistency in how the manual refers to language. Sometimes it is referred to "Julia" and sometimes "julia". I've updated the manual (and one place in the REPL help) such that when referring to Julia is always "Julia".

Also, the "Elementary Function" commit (d1cf43c) shouldn't be backported as it that is a 0.5 feature.

omus committed:
Disable syntax highlighting of `julia` options

omus committed:
Made splice example into a doctest

tkelman commented on doc/manual/parallel-computing.rst:
```diff
@@ -860,7 +860,7 @@ If ``io`` is not specified, ``host`` and ``port`` are used to connect.
 additional workers. ``count`` with an integer value ``n`` will launch a total of ``n`` workers,
 while a value of ``:auto`` will launch as many workers as cores on that machine.
-``exename`` is the name of the julia executable including the full path.
+``exename`` is the name of the Julia executable including the full path.
```
this might be better as quoted lowercase

omus replied:
I was thinking that "Julia" would refer the the language while "\`julia\`" refers to the command-line program. Does that seem sensible?

StefanKarpinski replied:
Yes, exactly. Probably makes sense to also put `julia` in backticks.

tkelman replied:
Agreed. "julia executable" is pretty clearly the latter.

tkelman replied:
not yet addressed here

omus committed:
Removed trailing colon in reference

Also prepended "see" to the references and fixed a triple space after
a period.

omus committed:
Remove unnecessary punctuation

omus committed:
More Julia over julia consistency

"Julia" refers to the language while "`julia`" (lowercase in backticks)
specifically refers to the executable.

kshyatt commented:
AV restarted. @tkelman @StefanKarpinski do the latest changes here look ok?

omus commented:
Feel free to merge this whenever. I have a couple more corrections to add but I can always make another PR.
